src/arrayMethodJoin.js

```diff
@@ -5,8 +5,45 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let strFromArr = '';
+    const separatorSign = checkSeparator(separator);
+
+    if (this.length === 0) {
+      return strFromArr;
+    }
```

Comment:
```suggestion
```
It's redundant. If `this.length === 0` then the loop will not even start:
`for (let i = 0; i < this.length; i++)` will be `for (let i = 0; i < 0; i++)`. Here 0 is never less than 0, so nothing will happen